Doppler.ContactPolicies.Data.Access/Repositories/ContactPoliciesSettings/ContactPoliciesSettingsRepository.cs

```diff
@@ -18,28 +18,25 @@ public ContactPoliciesSettingsRepository(IDatabaseConnectionFactory databaseConn
 
         #region public methods
 
-        public async Task<Entities.ContactPoliciesSettings> GetContactPoliciesSettingsAsync(string accountName)
+        public async Task<Entities.ContactPoliciesSettings> GetContactPoliciesSettingsAsync(int idUser)
         {
             using var connection = await _databaseConnectionFactory.GetConnection();
             const string query =
-                @"select usl.IdUser, usl.Active, usl.Interval [IntervalInDays], usl.Amount [EmailsAmountByInterval], u.Email [AccountName]
+                @"select convert(bit, (case when usl.IdUser is null then 0 else 1 end)) as UserHasContactPolicies, usl.Active, usl.Interval [IntervalInDays], usl.Amount [EmailsAmountByInterval], u.Email [AccountName]
                 from [User] u
                 left join [UserShippingLimit] usl on u.IdUser = usl.IdUser and usl.Enabled = 1
-                where u.Email = @Email;
+                where u.IdUser = @IdUser;
                 select sl.IdSubscribersList [Id], sl.Name
                 from [SubscribersListXShippingLimit] sls
                 inner join [SubscribersList] sl on sl.IdSubscribersList = sls.IdSubscribersList
-                inner join [User] u on u.IdUser = sl.IdUser where u.Email = @Email;";
+                inner join [User] u on u.IdUser = sl.IdUser where u.IdUser = @IdUser;";
```

Comment:
**I am not sure about that**
Do we still need this join? Maybe we can remove that and use the `IdUser` present in `SubscribersList` in the `where` condition:
```suggestion
                where sl.IdUser = @IdUser;";
```
I was testing that and the results were the same.

Reply:
👍